src/Mod/PartDesign/App/FeatureHelix.cpp

```diff
@@ -229,6 +234,12 @@ App::DocumentObjectExecReturn* Helix::execute()
 
         TopoDS_Shape face = sketchshape;
         face.Move(invObjLoc);
+
+        Bnd_Box bounds;
+        BRepBndLib::Add(path, bounds);
+        double size=sqrt(bounds.SquareExtent());
+        ShapeFix_ShapeTolerance fix;
+        fix.LimitTolerance(path, Precision::Confusion() * 1e-6 * size ); // needed to produce valid Pipe for very big parts
```

Comment:
You introduce a new hard coded parameter, can you explain the choice of the value?